lib/connect.js

```diff
@@ -81,7 +81,10 @@ module.exports = function connect (dht, publicKey, opts = {}) {
     relaySocket: null,
     relayClient: null,
     relayPaired: false,
-    relayKeepAlive: opts.relayKeepAlive || 5000
+    relayKeepAlive: opts.relayKeepAlive || 5000,
+
+    // Used by the server when tie-breaking between two connections from the same peer
+    timestamp: opts.timestamp ?? -1
```

Comment:
this default of `-1` makes the timestamp behavior _opt in_... which may be a pain-in-the-butt for most users, who'll not benefit from this change until they modify their code to pass `{ timestamp: Date.now() }` in the options to `connect`.

However, I tend to be very conservative when it comes to breaking changes, and my philosophy is usually to make all enhancements "opt in" until the next major version bump, at which time I update the defaults.

Reply:
just use 0 as the default, then the uint can be unsigned, and 0 is pretty appropriate for that. also means we can just do || 0